### DISCUSSION TITLE
Upgraded to Ghidra 1.2 and tried to upgrade GhidraDev in Eclipse, but its About says I still have old GhidraDev

### DISCUSSION THREAD

**Discussion by hippietrail** [NationalSecurityAgency/ghidra#6964] [Q&A]
After the upgrade:

In Eclipse's `Help > Install New Software... > What is already installed?`
<img width="712" alt="image" src="https://github.com/user-attachments/assets/e6e608bb-5bb9-44fe-a93e-e23f31f895c3">

But in Eclipse's `GhidraDev > About GhidraDev...` still shows:
<img width="579" alt="image" src="https://github.com/user-attachments/assets/b4afab5e-0c6c-45f0-a001-eb721a92be04">

What Eclipse's `GhidraDev > Link Ghidra...` shows
<img width="609" alt="image" src="https://github.com/user-attachments/assets/18ec2fd0-9a35-4584-952f-5ea23becb622">
Then clicking on the `+` button
<img width="726" alt="image" src="https://github.com/user-attachments/assets/ba9ef326-3a2d-44b0-9a39-d44fd2170953">

What happens when I try to use the `Add...` button to add Ghidra 11.2
<img width="579" alt="image" src="https://github.com/user-attachments/assets/1fa9be13-f167-48b4-80c3-e922722fe090">

It also won't let me install the new GhidraDev a second time and I can't see any way to uninstall either the old or new GhidraDev to try again.

**Comment by ryanmkurtz** [comment]
Did you restart Eclipse after doing the upgrade?

**Reply by hippietrail** [reply]
Yes. I think the upgrade prompted me to the first time, but I've restarted it a couple more times since then. Let me try again now... still no difference. Here's my Eclipse version details by the way:
Version: 2024-06 (4.32.0)
Build id: 20240606-1231

I'm on macOS 14.4

**Reply by ryanmkurtz** [reply]
Hmm...not sure what's going on.  Can you uninstall it, make sure the uninstall worked, and then install again rather than going through the upgrade path?

**Reply by hippietrail** [reply]
Ah I couldn't see how to uninstall GhidraDev before but now I found it. It told me it worked and it restarted Eclipse but while it's no longer in the `already installed` list, the GhidraDev menu is still there and its `About GhidraDev...` still tells me the old version is still installed.

**Reply by ryanmkurtz** [reply]
Hmmm, can you go to your eclipse installation on disk and check to see if it's in the `dropins/` directory?  If so, try deleting it from there.  When you let Ghidra install it for you, that's where it drops it.

**Reply by hippietrail** [reply]
There is a `/Applications/Eclipse.app/Contents/Eclipse/dropins` directory, but it is empty, not even any hidden files.

**Reply by ryanmkurtz** [reply]
If that's empty I'm really not sure why you'd still be seeing an old GhidraDev installed if it's not being listed in the Eclipse Installation Details.

**Reply by ryanmkurtz** [reply]
I just went through the installation on my Mac: 
<img width="565" alt="image" src="https://github.com/user-attachments/assets/68a34e8a-cc92-4033-a26f-b839b2caa8c2">

**Reply by ryanmkurtz** [reply]
You might have to install a fresh Eclipse to resolve the issue, at least in the short term.  I would assume that would fix the issue.
